fastapi/dependencies/utils.py

```diff
@@ -690,7 +693,7 @@ async def solve_dependencies(
 def _validate_value_with_model_field(
     *, field: ModelField, value: Any, values: Dict[str, Any], loc: Tuple[str, ...]
 ) -> Tuple[Any, List[Any]]:
-    if value is None:
+    if value is _NOT_SPECIFIED:
```

Comment:
```suggestion
    if value is _NOT_SPECIFIED or value == "":
```

Changing it to this also fixes my issue: https://github.com/fastapi/fastapi/discussions/13255

It looks like request_body_to_args was refactored to _validate_value_with_model_field in https://github.com/fastapi/fastapi/pull/12117/files but it dropped the empty string case.

Reply:
@tonyay163 Your issue seems unrelated though. The fact that you pass an empty string and it is parsed as an empty string (instead of `None`) sounds more like a feature than a bug...

Reply:
breaking changes are features? this worked in a previous version but does not anymore. how can this be used by so many people and have breaking changes from a refactor?

Reply:
You are right that the behaviour has changed, but for me the behaviour is correct now and was incorrect in the past. That's why I called it a feature, cause for me the behaviour was "buggy" before :)

Anyway, my opinion on your issue doesn't really matter. My point here was that your issue is "unrelated" to this PR. This PR is about broken functionality, about not being able to have optional Form fields at all.
How an optional Form field should be parsed is (imho) a different discussion.